### PR TITLE
vimc-6717: extract pop given demographic source 

### PR DESCRIPTION
**Pull request by xiangli313**
youtrack:
https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-6717

We were pulling population data for a given touchstone parameter. Now we want to go beyond that. Population data can also be extracted given demographic source.

### CODE REVIEW

File: R/commonly_used_db_data.R [hunk 1]
```diff
@@ -238,18 +240,27 @@ extract_vaccination_history <- function(con, touchstone_cov = "201710gavi", touc
 ##' @param country_ All countries if NULL. Or specify a vector of countries
 ##' @param year_ All years if NULL. Or specify a vector of years
 ##' @param age_ All age groups if NULL. Or specify a vector of age groups
+##' @param demographic_source one of demographic_source.code, this works for IU where demography is no a model run version
```
Comment (r-ash) [inline R/commonly_used_db_data.R]:
```suggestion
##' @param demographic_source one of demographic_source.code, this works for IU where demography is not a model run version
```
?

File: R/commonly_used_db_data.R [hunk 2]
```diff
@@ -53,14 +53,16 @@ get_touchstone_id <- function(con, touchstone) {
 ##' @param external_population_estimates The rationales are 1. we can use external population estimates if any and if necessary;
 ##' 2. demographic uncertainty not only affects models, but also FVPs. If we are to conduct sensitivity analysis on impact_by_year_of_vaccination, we need to vary population input for adjusting FVPs.
 ##' @param full_description TRUE if including scenario_descriptions (coverage estimates will be duplicated for scenarios); and FALSE if only providing coverage estimates
+##' @param demographic_source this is a 2nd option for getting population data from montagu
```
Comment (r-ash) [inline R/commonly_used_db_data.R]: I think the doc string for this arg should include info from `get_population` i.e. include some info about what this value actually needs to be. I think probably also worth putting a note of the two different modes of running into the function details after the title.

Probably also worth adding some validation at the start of this function which checks that only one of `touchstone_pop` or `demographic_source` is set and not both or neither. Returning a useful error message.

File: R/commonly_used_db_data.R [hunk 3]
```diff
@@ -238,18 +240,27 @@ extract_vaccination_history <- function(con, touchstone_cov = "201710gavi", touc
...
+  if(is.null(demographic_source)){
+    if (!grepl("-", touchstone_pop)) {
+      touchstone_pop <- get_touchstone(con, touchstone_pop)
+      message("touchstone version is not specified. Lateset version is used.")
+    }
+    ## this table get population data as you wish
+    sql <- read_sql(system_file("sql/population.sql"))
+    pop_src <- touchstone_pop
+  } else {
+    sql <- read_sql(system_file("sql/population2.sql"))
```
Comment (r-ash) [inline R/commonly_used_db_data.R]: Minor point but you could update these to a more meaningful file name. Something like `touchstone_population.sql` and `demographic_population.sql` or something similar

**Review by r-ash — CHANGES_REQUESTED** [review]
Impl wise this looks good but I think worth tidying up some of the docs and maybe adding some validation as the interface to this function is already confusing and putting some small guard in place I think will make it much easier to use

File: R/commonly_used_db_data.R (hunk 2 above)
Comment (xiangli313) [reply R/commonly_used_db_data.R]: The two parameters are hierachical.
  when touchstone_pop is null, touchstone_cov is used for touchstone_pop
  when demographic_source is not null, touchstone_pop is not used but ruled out
I have added a few lines of codes to explain and validate.

File: R/commonly_used_db_data.R (hunk 3 above)
Comment (xiangli313) [reply R/commonly_used_db_data.R]: Done.

**Review by r-ash — APPROVED** [review]

**Pull request merged**